pkg/dbutil/common.go

```diff
@@ -780,3 +833,16 @@ func GetParser(sqlModeStr string) (*parser.Parser, error) {
 	parser2.SetSQLMode(sqlMode)
 	return parser2, nil
 }
+
+// GetParserForDB discovers ANSI_QUOTES in db's session variables and returns a proper parser
+func GetParserForDB(db *sql.DB) (*parser.Parser, error) {
+	ansiQuotes, err := HasAnsiQuotesMode(db)
+	if err != nil {
+		return nil, err
+	}
+	sqlMode := ""
+	if ansiQuotes {
+		sqlMode = "ANSI_QUOTES"
+	}
+	return getParser(sqlMode)
```

Comment:
Can we set `sql_mode` for all modes from `GetSQLMode` rather than only `ANSI_QUOTES`?

Reply:
https://github.com/pingcap/tidb-tools/pull/381/commits/f70d144ff85a84a3b31bb29cdf34db015327b069

Okay, going to align DM's parser to this behaviour